cmd/bosun/database/state_data.go

```diff
@@ -93,16 +96,80 @@ func (d *dataAccess) GetRenderedTemplates(incidentId int64) (*models.RenderedTem
 	defer conn.Close()
 
 	b, err := redis.Bytes(conn.Do("GET", renderedTemplatesKey(incidentId)))
+	renderedT := &models.RenderedTemplates{}
 	if err != nil {
+		if err == redis.ErrNil {
+			return renderedT, nil
+		}
 		return nil, slog.Wrap(err)
 	}
-	renderedT := &models.RenderedTemplates{}
 	if err = json.Unmarshal(b, renderedT); err != nil {
 		return nil, slog.Wrap(err)
 	}
 	return renderedT, nil
 }
 
+func (d *dataAccess) GetRenderedTemplateKeys() ([]string, error) {
+	conn := d.Get()
+	defer conn.Close()
+
+	//ledis uses XSCAN cursor "KV" MATCH foo
+	//redis uses SCAN cursor MATCH foo
+	cmd := "SCAN"
```

Comment:
Maybe Redis and Ledis SCAN could be a function outside of this that can be reused (and make this func shorter)

Reply:
Yeah, we did that with some others, but the re-ordering of the args makes it tricky. I will factor out.